### PR TITLE
fix(protocol): fix signature reuse bug

### PR DESCRIPTION
**Pull request by dantaik**

### CODE REVIEW

**Comment by adaki2004** [comment]
Well, it is OK, but i dont see any security concern here, because basically it is just a "delegation".

So Bob delegates to Alice to withdran on behalf of him, but with this solution it guards it with a nonce (so one-time delegation) ? Ok, maybe enhanced security, so I'm OK with it.

File: packages/protocol/contracts/team/TimelockTokenPool.sol [hunk 1]
```diff
@@ -169,18 +174,25 @@ contract TimelockTokenPool is EssentialContract, EIP712Upgradeable {
...
+        _withdraw(account, _to);
     }
 
     /// @notice Gets the hash to be signed to authorize an withdrawal.
     /// @param _to The destination address.
+    /// @param _nonce The withdrawal nonce.
     /// @return The hash to be signed.
-    function getWithdrawalHash(address _to) public view returns (bytes32) {
-        return _hashTypedDataV4(keccak256(abi.encode(TYPED_HASH, _to)));
+    function getWithdrawalHash(address _to, uint256 _nonce) public view returns (bytes32) {
```
Comment (adaki2004) [inline packages/protocol/contracts/team/TimelockTokenPool.sol:194]: to be cross-chain fixed, maybe block.chainId (?).

Comment (dantaik) [reply packages/protocol/contracts/team/TimelockTokenPool.sol:194]: its part of _hashTypedDataV4 I believe.

Comment (adaki2004) [reply packages/protocol/contracts/team/TimelockTokenPool.sol:194]: > its part of _hashTypedDataV4 I believe.

Ah yeah, false alarm!

**Review by adaki2004 — APPROVED** [review]

**Review by Brechtpd — APPROVED** [review]
I guess the owner may indeed want to stop using the `to` address.

**Pull request merged**